### PR TITLE
ci(jenkins): transform jUnit output for test stages

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Generates test output 

## Why is it important?

Visualise the test output when running the test stage.

## Related issues

Closes https://github.com/elastic/apm-agent-go/issues/712

### CODE REVIEW

**Comment by robgil** [comment]
# [Codecov](https://codecov.io/gh/elastic/apm-agent-go/pull/714?src=pr&el=h1) Report
> Merging [#714](https://codecov.io/gh/elastic/apm-agent-go/pull/714?src=pr&el=desc) into [master](https://codecov.io/gh/elastic/apm-agent-go/commit/a983a0c705c8ca678d4557b8e844802767548a44&el=desc) will **decrease** coverage by `0.02%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/elastic/apm-agent-go/pull/714?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #714      +/-   ##
==========================================
- Coverage   84.64%   84.61%   -0.03%     
==========================================
  Files         127      127              
  Lines        8276     8276              
==========================================
- Hits         7005     7003       -2     
- Misses        889      891       +2     
  Partials      382      382              
```


| [Impacted Files](https://codecov.io/gh/elastic/apm-agent-go/pull/714?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [tracer.go](https://codecov.io/gh/elastic/apm-agent-go/pull/714/diff?src=pr&el=tree#diff-dHJhY2VyLmdv) | `87.36% <0.00%> (-0.95%)` | :arrow_down: |
| [transport/http.go](https://codecov.io/gh/elastic/apm-agent-go/pull/714/diff?src=pr&el=tree#diff-dHJhbnNwb3J0L2h0dHAuZ28=) | `85.84% <0.00%> (+1.23%)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/elastic/apm-agent-go/pull/714?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/elastic/apm-agent-go/pull/714?src=pr&el=footer). Last update [837aa09...6133c5e](https://codecov.io/gh/elastic/apm-agent-go/pull/714?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: scripts/jenkins/build-test.sh [hunk 1]
```diff
@@ -5,4 +5,16 @@ set -euxo pipefail
...
+
+# Run the tests
+set +e
+export OUT_FILE="build/test-report.out"
+mkdir -p build
+make test 2>&1 | tee ${OUT_FILE}
+status=$?
+
+go get -v -u github.com/jstemmer/go-junit-report
+go-junit-report > "build/junit-apm-agent-go-${GO_VERSION}.xml" < ${OUT_FILE}
```
Comment (mdelapenya) [inline scripts/jenkins/build-test.sh]: Isn't this similar to what we do here: https://github.com/elastic/apm-agent-go/blob/master/scripts/docker-test.sh#L19?

Comment (v1v) [reply scripts/jenkins/build-test.sh]: Indeed

Comment (v1v) [reply scripts/jenkins/build-test.sh]: In other words, the current JUnit reporting is only enabled for the coverage that for some reason uses `docker-compose`. While the test stage uses the CI Workers context instead a docker container.

For such, the implementation details are different, that particular `docker-test.sh`, it was created to align the same steps for travis and jenkins: 
- https://github.com/elastic/apm-agent-go/pull/588

Comment (mdelapenya) [reply scripts/jenkins/build-test.sh]: So at some point, when we run this only on Jenkins, we'll align scripts I guess

**Comment by mdelapenya** [comment]
Sorry, the robot in me approved and merged as a robot ...

**Pull request merged**